backend/src/main/java/com/github/redstylzz/backend/service/PaymentService.java

```diff
@@ -30,44 +34,60 @@ private boolean paymentExists(String paymentID) {
         return paymentRepo.existsByPaymentID(paymentID);
     }
 
+    private void calculatePaymentSum(String userID, String categoryID) {
+        List<Payment> payments = paymentRepo.getAllByUserIDAndCategoryID(userID, categoryID);
+        BigDecimal sum = payments.stream().map(Payment::getAmount).reduce(BigDecimal.ZERO, BigDecimal::add);
+        categoryService.setCategorySum(userID, categoryID, sum);
+    }
+
+    private List<PaymentDTO> getPaymentAsDTO(String userID, String categoryID) {
+        return paymentRepo
+                .getAllByUserIDAndCategoryID(userID, categoryID)
+                .stream()
+                .map(Payment::convertPaymentToDTO)
+                .toList();
+    }
 
-    public List<Payment> getPayments(String userID, String categoryID) {
-        return paymentRepo.getAllByUserIDAndCategoryID(userID, categoryID);
+    public List<PaymentDTO> getPayments(String userID, String categoryID) {
+        return getPaymentAsDTO(userID, categoryID);
     }
 
-    public List<Payment> addPayment(String userID, Payment payment) throws CategoryDoesNotExistException {
+    public List<PaymentDTO> addPayment(String userID, Payment payment) throws CategoryDoesNotExistException {
         if (categoryExistent(userID, payment.getCategoryID())) {
             payment.setPaymentID(UUID.randomUUID().toString());
             payment.setUserID(userID);
             payment.setSaveDate(new Date());
             paymentRepo.save(payment);
+            calculatePaymentSum(userID, payment.getCategoryID());
         } else {
             throw new CategoryDoesNotExistException();
         }
-        return paymentRepo.getAllByUserIDAndCategoryID(userID, payment.getCategoryID());
+        return getPaymentAsDTO(userID, payment.getCategoryID());
     }
 
-    public List<Payment> deletePayment(String userID, Payment payment) throws PaymentDoesNotExistException, CategoryDoesNotExistException {
-        if (categoryExistent(userID, payment.getCategoryID())) {
```

Comment:
categoryExistent is not a good method name, a method should include a verb indicating what interaction it carries out or should be a yes/no question to indicate that the return type is a boolean (e.g. isCategoryExistent)